tasktiger/task.py

```diff
@@ -287,6 +296,13 @@ def delay(self, when=None):
         # ensure there are no serialization errors.
         serialized_task = json.dumps(self._data)
 
+        if max_queue_size:
+            # This will fail adding a unique task that already is queued but
+            # the queue size is at the max
+            queue_size = tiger.get_total_queue_size(self.queue)
```

Comment:
I'm late to this one, but it would be nice if this was done as part of the pipeline. E.g. abort the pipeline if the queue is full. Rather than doing a round trip before queueing each task, which is also prone to race conditions (though it's no big deal).